### PR TITLE
agent toggles and meters into one row 

### PR DESCRIPTION
**Pull request by michaelkirk**
also, fixed the horizontal jitter by giving the toggle buttons a fixed width. 

https://user-images.githubusercontent.com/217057/111382711-8ddf3800-8664-11eb-83d0-7ca8a5b40b0f.mp4

Screen Shot 2021-03-16 at 2 28 24 PM

Screen Shot 2021-03-16 at 2 29 07 PM

The numbers start to look a little cramped as we get into 5 digits, but it's still legible, and I don't think this is super common. We could add more space, but there is a counter tension where more space makes the low numbers seem *too* spaced out.
Screen Shot 2021-03-16 at 2 30 13 PM

Screen Shot 2021-03-16 at 2 30 53 PM
Screen Shot 2021-03-16 at 2 30 44 PM
Screen Shot 2021-03-16 at 2 30 29 PM

### CODE REVIEW

**Comment by michaelkirk** [comment]
I forgot to include screenshots of the tooltips:

Screen Shot 2021-03-16 at 2 44 41 PM
Screen Shot 2021-03-16 at 2 44 50 PM

File: widgetry/src/geom/geom_batch_stack.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    Vertical,
+}
+
+/// Similar to [`Widget::row`]/[`Widget::column`], but for [`GeomBatch`]s instead of [`Widget`]s,
+/// and follows a builder pattern
+///
+/// You can add items incrementally, change `spacing` and `axis`, and call `batch` at the end to
+/// apply these rules to produce an aggeregate `GeomBatch`.
+#[derive(Debug)]
+pub struct GeomBatchStack {
```
Comment (michaelkirk) [inline widgetry/src/geom/geom_batch_stack.rs:15]: moved this from the ButtonBuilder and made it public.

File: widgetry/src/widgets/image.rs [hunk 2]
```diff
@@ -2,18 +2,43 @@ use crate::{
...
+#[derive(Clone, Debug)]
 pub struct Image<'a> {
-    filename: &'a str,
+    source: ImageSource<'a>,
     tooltip: Option<Text>,
     color: Option<RewriteColor>,
 }
 
+#[derive(Clone, Debug)]
+pub enum ImageSource<'a> {
```
Comment (michaelkirk) [inline widgetry/src/widgets/image.rs:15]: Moved this from `ButtonBuilder` too.

**Comment by dabreegster** [comment]
The code changes are fantastic, and I love the combined design and lack of jitter. The amount of unused space when zoomed in is kind of jarring, though. I tried out a few alternatives:
wider
columns
But neither looks better to me. I'm tempted to just go with the extra space for now, and maybe use it for something in the future, or adjust the minimap to be wider (but not any taller).

Any thoughts?

**Comment by michaelkirk** [comment]
image

This one seems promising! 

I think it might look a little better if you aligned the left edge of the "car" and "bus" checkbox with the left edge of the minimap.

image

**Comment by michaelkirk** [comment]
A couple other thoughts which I don't think require immediate action:

1. the minimap doesn't *have* to be square.
2. the "zoom" controls for the minimap seem to have pretty marginal utility, at least for me.

**Comment by dabreegster** [comment]
I vote merging this as it is, and we can keep experimenting -- maybe I'll ask for a vote in the Slack.

> I think it might look a little better if you aligned the left edge of the "car" and "bus" checkbox with the left edge of the minimap.

Hmm, yeah, that would look decent. I'll send a screenshot.

> the minimap doesn't have to be square.

Right -- a wider rectangle would nicely fill up the extra space.

> the "zoom" controls for the minimap seem to have pretty marginal utility, at least for me.

I agree for small maps, but on `huge_seattle` or `ballard`, you can barely see anything at the default level. Zooming in a bit can help there.

**Pull request merged**